src/Integration/Suppression/SonarQubeIssuesProvider.cs

```diff
@@ -0,0 +1,106 @@
+
+namespace SonarLint.VisualStudio.Integration.Suppression
+{
+    public sealed class SonarQubeIssuesProvider : ISonarQubeIssuesProvider, IDisposable
+    {
+        private const double MillisecondsToWaitBetweenRefresh = 1000 * 60 * 60 * 1; // 1 hour
+
+        private readonly IHost host;
+        private readonly System.Timers.Timer refreshTimer;
+        private readonly IActiveSolutionBoundTracker solutionBoundTacker;
+
+        private IList<SonarQubeIssue> cachedSuppressedIssues;
+        private bool isDisposed;
+        private CancellationTokenSource cancellationTokenSource;
+
+        [ImportingConstructor]
+        internal SonarQubeIssuesProvider(IHost host, IActiveSolutionBoundTracker solutionBoundTacker)
+        {
+            // Ideally we would have the bound project as part of IActiveSolutionBoundTracker.SolutionBindingChanged event
+            // args and we could get rid of the dependency to IHost here.
+            this.host = host;
+            this.solutionBoundTacker = solutionBoundTacker;
+            this.solutionBoundTacker.SolutionBindingChanged += OnSolutionBoundChanged;
+
+            // TODO: Use mockable timer
+            refreshTimer = new System.Timers.Timer { AutoReset = true, Interval = MillisecondsToWaitBetweenRefresh };
+            refreshTimer.Elapsed += OnRefreshTimerElapsed;
+        }
+
+        public void Dispose()
+        {
+            if (this.isDisposed)
+            {
+                return;
+            }
+
+            refreshTimer.Dispose();
+            this.isDisposed = true;
+        }
+
+        public IEnumerable<SonarQubeIssue> GetSuppressedIssues(string projectGuid, string filePath)
+        {
+            // TODO: Block the call while the cache is being built
+            return this.cachedSuppressedIssues.Where(x => x.FilePath == filePath && x.ModuleKey == BuildModuleKey(projectGuid));
+        }
+
+        private string BuildModuleKey(string projectGuid)
+        {
+            var sonarQubeProjectKey = this.host.VisualStateManager.BoundProjectKey;
+            return $"{sonarQubeProjectKey}:{sonarQubeProjectKey}:{projectGuid}";
+        }
+
+        private async void OnRefreshTimerElapsed(object sender, System.Timers.ElapsedEventArgs e)
+        {
+            await SynchronizeSuppressedIssues();
+        }
+
+        private async void OnSolutionBoundChanged(object sender, bool e)
+        {
+            if (solutionBoundTacker.IsActiveSolutionBound)
+            {
+                await SynchronizeSuppressedIssues();
+                refreshTimer.Start();
+            }
+            else
+            {
+                refreshTimer.Stop();
+            }
+        }
+
+        private async Task SynchronizeSuppressedIssues()
+        {
+            cancellationTokenSource?.Cancel();
+            cancellationTokenSource = new CancellationTokenSource();
+            this.cachedSuppressedIssues = await this.host.SonarQubeService.GetSuppressedIssuesAsync(
```

Comment:
Might want to code defensively here to avoid a potential race condition: fetch the _BoundProjectKey_ first and check it is valid (in case the _VisualStateManager_ has changed in the meantime on another thread).